File: gluecodium/src/test/resources/smoke/nesting/output/android/com/example/smoke/UseFreeTypes.java
/*
 *
 */
package com.example.smoke;
import android.support.annotation.NonNull;
import com.example.NativeBase;
import java.util.Date;
public final class UseFreeTypes extends NativeBase {
    /**
     * For internal use only.
     * @hidden
     */
    protected UseFreeTypes(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
    @NonNull
    public native Date doStuff(@NonNull final FreePoint point, @NonNull final FreeEnum mode) throws FreeException;
}

File: gluecodium/src/test/resources/smoke/nesting/output/android-kotlin/com/example/smoke/UseFreeTypes.kt
/*

 *
 */

@file:JvmName("UseFreeTypes")

package com.example.smoke

import com.example.NativeBase
import java.util.Date

class UseFreeTypes : NativeBase {



    /**
     * For internal use only.
     * @suppress
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}




    @Throws(FreeException::class)
    external fun doStuff(point: FreePoint, mode: FreeEnum) : Date




    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
    }
}
